### PR TITLE
Add publication to PyPI

### PR DESCRIPTION
**Pull request by jvegreg**
We are having troube publishing ESMValCore to PyPI because we are referencing this project by url. This PR is to get this project available on PyPI

### CODE REVIEW

**Comment by bouweandela** [comment]
Made an issue for it here: ESMValGroup/ESMValCore#997

File: .github/workflows/action-pypi-build-and-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  release:
+    types: [published]
+  # use this for testing
+  push:
+    branches:
+      - deploy_pypi
+
+jobs:
+  build-n-publish:
+    name: Build and publish ESMValCore on PyPi
```
Comment (valeriupredoi) [inline .github/workflows/action-pypi-build-and-deploy.yml]:
```suggestion
    name: Build and publish ESMValTool Sample Data package on PyPi
```

**Review by valeriupredoi — APPROVED** [review]

File: setup.py [hunk 2]
```diff
@@ -11,7 +11,8 @@
     name='ESMValTool sample data',
```
Comment (bouweandela) [inline setup.py]:
```suggestion
    name='ESMValTool_sample_data',
```

File: .github/workflows/action-pypi-build-and-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+name: PyPi Build and Deploy 🐍📦
+
+on:
+  release:
+    types: [published]
+  # use this for testing
+  push:
+    branches:
+      - deploy_pypi
```
Comment (bouweandela) [inline .github/workflows/action-pypi-build-and-deploy.yml]:
```suggestion
  push:
    branches:
      - master
```

Comment (bouweandela) [reply .github/workflows/action-pypi-build-and-deploy.yml]: For testing that the build succeeds, upload will only happen on tags (see line 41)

**Pull request merged**